examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+        # dependency is accounted for by changing the pixel values (in electrons) and applying
+        # a constant nominal gain later, which is unity in our demo.
+
+        # Save the image before applying the transformation to see the difference:
+        final_image_3 = final_image.copy()
+
+        NLfunc = wfirst.NLfunc        # a quadratic non-linear function
+        final_image.applyNonlinearity(NLfunc=NLfunc)
+        logger.debug('Applied nonlinearity to {0}-band image'.format(filter_name))
+        final_image_4 = final_image.copy()
+        # Isolate the changes due to non-linear gain.
+        diff = final_image_4-final_image_3
+
+        out_filename = os.path.join(outpath,'demo13_NL_{0}.fits'.format(filter_name))
+        final_image_4.write(out_filename)
+        out_filename = os.path.join(outpath,'demo13_diff_NL_{0}.fits'.format(filter_name))
+        diff.write(out_filename)
+
+        # 4) Including Interpixel capacitance:
+        # The voltage read at a given pixel location is influenced by the charges present in
+        # the neighboring pixel locations due to capacitive coupling of sense nodes. This
+        # interpixel capacitance effect is modeled as a linear effect that is described as a
+        # convolution of a 3x3 kernel with the image. The WFIRST kernel is not normalized to
+        # have the entries add to unity and hence must be normalized inside the routine.
+
+        final_image.applyIPC(IPC_kernel=wfirst.ipc_kernel,edge_treatment='extend',
+                             kernel_normalization=True)
+        # Here, we use `edge_treatment='extend'`, which pads the image with zeros before
+        # applying the kernel. The central part of the image is retained.
+        logger.debug('Applied interpixel capacitance to {0}-band image'.format(filter_name))
+        final_image_5 = final_image.copy()
+        # Isolate the changes due to the interpixel capacitance effect.
+        diff = final_image_5-final_image_4
+
+        out_filename = os.path.join(outpath,'demo13_IPC_{0}.fits'.format(filter_name))
+        final_image_5.write(out_filename)
+        out_filename = os.path.join(outpath,'demo13_diff_IPC_{0}.fits'.format(filter_name))
+        diff.write(out_filename)
+
+        # 5) Adding read noise:
+        # Read noise is the noise due to the on-chip amplifier that converts the charge into an
+        # analog voltage.
+        read_noise = galsim.CCDNoise(rng)
+        read_noise.setReadNoise(wfirst.read_noise)
+        final_image.addNoise(read_noise)
+        final_image_6 = final_image.copy()
+
+        logger.debug('Added readnoise to {0}-band image'.format(filter_name))
+
+        # Technically we have to apply the gain, dividing the signal in e- by the voltage gain
+        # in e-/ADU to get a signal in ADU.  For WFIRST the gain is expected to be around 1,
+        # so it doesn't really matter, but for completeness we include this step.
```

Comment:
Probably shouldn't be so flippant here about the gain when you are being so careful about all these other effects.  Gains for real detectors are almost never exactly 1, even if they aim to make them approximately 1.  Typically gains vary from one amp to another etc.  So the text here should probably be more along the lines of:

We divide by the gain to convert from e- to ADU.  Currently, the gain value in the wfirst module is just set to 1, since we don't know what the exact gain will be, although it is expected to be approximately 1.  Eventually, this may change when the camera is finished being assembled, and there may be a different value for each SCA.  For now, there is just a single number, which is equal to 1.